packages/@sanity/portable-text-editor/src/editor/plugins/createWithPatches.ts

```diff
@@ -237,8 +230,12 @@ export function createWithPatches({
         // Do nothing
       }
 
-      // Unset the value if the operation made the editor empty
-      if (editorIsEmpty && ['remove_text', 'remove_node'].includes(operation.type)) {
+      // Unset the value if a operation made the editor empty
+      if (
+        !editorWasEmpty &&
+        editorIsEmpty &&
+        ['set_node', 'remove_text', 'remove_node'].includes(operation.type)
```

Comment:
The unset that previously was needed on line 176 is now taken care of here instead, which is much safer.